runtime-tools-quarkus-extension-parent/runtime/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/forms/impl/FormsStorageImpl.java

```diff
@@ -37,39 +54,120 @@
-        if (formsFolder == null) {
-            return 0;
-        }
-
-        File rootFolder = FileUtils.toFile(formsFolder);
+        return formBaseInfoCollection.keySet().size();
+    }
 
-        Collection<File> forms = readForms();
+    @Override
+    public Collection<FormInfo> getFormInfoList(FormFilter names) {
+        final Collection<FormInfo> formInfoCollection = new ArrayList();
+        Collection<String> matchedNames = null;
+        if (names.getNames().size() > 0) {
+            matchedNames = formBaseInfoCollection.keySet().stream().filter(name -> names.getNames().contains(name)).collect(Collectors.toList());
+        } else {
+            matchedNames = formBaseInfoCollection.keySet();
+        }
 
-        forms.forEach(form -> {
-            System.out.println(form.getName());
+        matchedNames.forEach(name -> {
+            Collection<FormInfo> formBaseInfos = formBaseInfoCollection.get(name);
+            formBaseInfos.forEach(f -> {
+                if (!f.getType().equals(FormInfo.FormType.CONFIG)) {
+                    formInfoCollection.add(f);
+                }
+            });
         });
+        return formInfoCollection;
+    }
 
-        return forms.size();
+    private FormInfo.FormType getFormType(String type) {
+        switch (type) {
+            case "html":
+                return FormInfo.FormType.HTML;
+            case "tsx":
+                return FormInfo.FormType.TSX;
+            case "config":
+                return FormInfo.FormType.CONFIG;
+        }
+        return null;
     }
 
     @Override
-    public Collection<FormInfo> getFormList() {
-        return readForms().stream()
-                .map(file -> new FormInfo(FilenameUtils.removeExtension(file.getName()), FilenameUtils.getExtension(file.getName()), new Date(file.lastModified())))
-                .collect(Collectors.toList());
+    public Form getFormContent(String formName) throws IOException {
+        File formFile = getFormFile(formName);
+        File formConfig = getFormConfigFile(formName);
+        FormConfiguration formConfiguration = null;
+        if (formConfig != null && formConfig.exists()) {
+            formConfiguration = new FormConfiguration(IOUtils.toString(new FileInputStream(formConfig), Charset.forName("UTF-8")));
+        } else {
+            throw new FileNotFoundException("{0} dose not found");
+        }
+        Form form = new Form(formConfiguration);
+        if (formFile != null && formFile.exists()) {
+            form.setSource(IOUtils.toString(new FileInputStream(formFile), Charset.forName("UTF-8")));
+        } else {
+            throw new FileNotFoundException("{0}'s config file dose not found");
+        }
+        return form;
     }
 
-    private Collection<File> readForms() {
-        File rootFolder = FileUtils.toFile(formsFolder);
+    private File getFormFile(String formName) {
+        Collection<FormInfo> formBaseInfo = formBaseInfoCollection.get(formName);
```

Comment:
It should be a single FormInfo, from there we get the type and with the type we can resolve the form file (being tsx or html) to load the content-